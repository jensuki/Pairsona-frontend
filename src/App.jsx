import { useEffect } from 'react';
import Navbar from './components/Navbar';
import { UserProvider } from './context/UserContext';
import AppRoutes from './Routes';
import useAuth from './hooks/useAuth';
import useNotifications from './hooks/useNotifications';
import Loading from './components/Loading'
import AOS from 'aos';
import 'aos/dist/aos.css'; // aos styles
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import './styles/App.css';

/** main app comopnent for pairsona */

const App = () => {
  const {
    currUser,
    setCurrUser,
    token,
    signup,
    login,
    logout,
    loading
  } = useAuth();

  const {
    hasNotifications,
    hasNewMessages,
    checkNotifications
  } = useNotifications(currUser);


  // initialize aos once on app load
  useEffect(() => {
    AOS.init({
      duration: 1000,
      once: true,
    });
  }, []);

  // display loading spinner while fetching user data
  if (loading) return <Loading />

  return (
    <div className="App">
      <ToastContainer
        position="top-center" />
      <UserProvider
        value={{
          currUser,
          setCurrUser,
          token,
          signup,
          login,
          logout,
          loading,
          hasNotifications,
          hasNewMessages,
          checkNotifications,
        }}
      >
        <Navbar
          currUser={currUser}
          logout={logout}
          hasNotifications={hasNotifications}
          hasNewMessages={hasNewMessages}
        />
        <AppRoutes currUser={currUser} signup={signup} login={login} />
      </UserProvider>
    </div>
  );
};

export default App;
