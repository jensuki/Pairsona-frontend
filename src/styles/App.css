.Homepage,
.SignupForm,
.ResultsPage,
.LoginForm,
.Messages,
.QuizPage,
.ProfileForm,
.matches-prompt {
  margin: 50px auto;
  padding: 40px;
  box-shadow: 0 4px 10px var(--box-shadow);
  border-radius: var(--border-radius-md);
  background-color: var(--neutral-bg);
  backdrop-filter: blur(8px);
  opacity: .95;
}

h2 {
  margin: 40px 20px;
}

.cards {
  display: flex;
  flex-wrap: wrap;
  gap: 20px;
  justify-content: center;
  margin-top: 20px;
}

.card {
  margin: 10px;
  border-radius: var(--border-radius-md);
  padding: 20px;
  width: 250px;
  box-shadow: 0 4px 8px 2px var(--box-shadow);
  text-align: left;
  transition: transform 0.2s;
  background-color: var(--neutral-bg);
  backdrop-filter: blur(8px);
  opacity: .95;

  &:hover {
    transform: translateY(1px);
  }
}

.card h1 {
  margin-top: 0;
  font-size: 1.5rem;
}

.card p {
  margin: 10px 0;
  font-size: var(--font-size-base);
}

/* global forms */

.form-group label {
  display: flex;
  font-weight: var(--font-bold);
  margin-bottom: 10px;
}

.form-group {
  margin-bottom: 15px;
}

.form-group input {
  width: calc(100% - 17px);
  padding: 8px;
  font-size: var(--font-size-base);
  border: 1px solid #ccc;
  border-radius: var(--border-radius-sm);
}

.error,
.form-error {
  color: red;
  text-align: center;
  margin-top: 10px;
}

.success {
  color: green;
  text-align: center;
  margin-top: 10px;
}