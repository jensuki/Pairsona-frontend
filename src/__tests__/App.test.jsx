import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';
import { describe, test, expect } from 'vitest';

describe('App component', () => {
    test('renders without crashing', () => {
        render(
            <MemoryRouter>
                <App />
            </MemoryRouter>
        );
        expect(screen.getByText(/Authentic/i)).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
                <App />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
