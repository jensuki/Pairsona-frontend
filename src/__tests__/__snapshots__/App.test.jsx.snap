// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <section
      aria-atomic="false"
      aria-label="Notifications Alt+T"
      aria-live="polite"
      aria-relevant="additions text"
      class="Toastify"
    />
    <nav
      class="Navbar"
    >
      <div
        class="Navbar-brand"
      >
        <a
          aria-label="Home"
          href="/"
        >
          <img
            alt="Pairsona-Logo"
            class="Navbar-logo"
            src="/src/assets/logos/pairsona_logo.png"
          />
        </a>
      </div>
      <ul
        class="Navbar-links"
      >
        <li
          class="nav-item"
        >
          <a
            class=""
            data-discover="true"
            href="/login"
          >
            Login
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class=""
            data-discover="true"
            href="/signup"
          >
            Sign Up
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="Homepage"
    >
      <section
        class="hero"
        data-aos="fade-up"
      >
        <h1>
          Authentic Connections
          <br />
           For Every Journey
        </h1>
        <p>
          Meet like-minded people who align with your personality and values. Whether it's love, friendship, or meaningful chats, your connections - your way.
        </p>
        <a
          class="cta-button"
          data-discover="true"
          href="/quiz"
        >
          Take your MBTI Personality Quiz
        </a>
        <img
          alt="Guests Home avatar"
          class="logo-lg"
          src="/src/assets/logos/pairsona_logo_large.png"
        />
      </section>
      <hr />
      <section
        class="how-it-works"
        data-aos="fade-left"
      >
        <h2>
          How It Works
        </h2>
        <div
          class="steps"
        >
          <div
            class="step"
          >
            <h3>
              1. Sign Up
            </h3>
            <p>
              Create an account and join our community.
            </p>
          </div>
          <div
            class="step"
          >
            <h3>
              2. Take the Quiz
            </h3>
            <p>
              Discover your MBTI type with our simple quiz.
            </p>
          </div>
          <div
            class="step"
          >
            <h3>
              3. Get Matched
            </h3>
            <p>
              Find compatible people based on personality.
            </p>
          </div>
          <div
            class="step"
          >
            <h3>
              4. Connect
            </h3>
            <p>
              Connect and get to know your matches through our chat!
            </p>
          </div>
        </div>
      </section>
      <hr />
      <section
        class="features"
        data-aos="fade-up"
      >
        <h2>
          Why Choose Us?
        </h2>
        <ul>
          <li>
            <strong>
              Personality-Based Matches
            </strong>
             – We pair you based on MBTI compatibility.
          </li>
          <li>
            <strong>
              Flexible Connections
            </strong>
             – Looking for friends or something more? Your choice.
          </li>
          <li>
            <strong>
              Privacy-Focused
            </strong>
             – Your data, your control.
          </li>
        </ul>
      </section>
      <hr />
      <section
        class="testimonials"
      >
        <blockquote>
          This app made it easy to connect with fellow introverts in a discreet and comfortable way.
        </blockquote>
        <blockquote>
          A welcoming space with plenty of opportunities to form genuine friendships.
        </blockquote>
      </section>
      <section
        class="final-cta"
      >
        <h2>
          Ready to meet your match?
        </h2>
        <a
          class="cta-button"
          data-discover="true"
          href="/signup"
        >
          Get Started
        </a>
        <p
          class="login-link"
        >
          Already have an account? 
          <a
            data-discover="true"
            href="/login"
          >
            Log in
          </a>
        </p>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
